ql/ql/src/queries/performance/UnusedField.ql

```diff
@@ -0,0 +1,36 @@
+/**
+ * @name UnusedField
+ * @description A field that is not used in the characteristic predicate will contain every value
+ *              of its type when accessed in other predicates, which is probably not intended.
+ * @kind problem
+ * @problem.severity warning
+ * @id ql/unused-field
+ * @precision high
+ */
+
+import ql
+
+from ClassType clz, ClassType implClz, FieldDecl field, string extraMsg
+where
+  clz.getDeclaration().getAField() = field and
+  implClz.getASuperType*() = clz and
+  // The field is not accessed in the charpred (of any of the classes)
+  not exists(FieldAccess access |
+    access.getEnclosingPredicate() =
+      [clz.getDeclaration().getCharPred(), implClz.getDeclaration().getCharPred()]
```

Comment:
isn't `clz` implied by `implClz`, as you're using the `*` operator in L16?

Reply:
If not, this can be `[clz, implClz].getDeclaration().getCharPred()`, right?

Reply:
> isn't `clz` implied by `implClz`, as you're using the * operator in L16?

Not necessarily. There are other restrictions in the query that makes it so `clz` and `implClz` must sometimes be different classes. (I don't have any examples to link to though).  

> If not, this can be [clz, implClz].getDeclaration().getCharPred(), right? 

Yes it can. I'll do that.